File: frontend/src/services/unifiedErrorHandler.ts
/**
 * 统一错误处理系统
 * 整合所有错误处理机制，提供一致的错误处理体验
 */

import { message, notification } from 'antd'
import { AxiosError } from 'axios'

// 错误类型枚举
export enum ErrorType {
  NETWORK_ERROR = 'NETWORK_ERROR',
  API_ERROR = 'API_ERROR',
  VALIDATION_ERROR = 'VALIDATION_ERROR',
  AUTHENTICATION_ERROR = 'AUTHENTICATION_ERROR',
  AUTHORIZATION_ERROR = 'AUTHORIZATION_ERROR',
  BUSINESS_ERROR = 'BUSINESS_ERROR',
  INTERNAL_SERVER_ERROR = 'INTERNAL_SERVER_ERROR',
  UNKNOWN_ERROR = 'UNKNOWN_ERROR'
}

// 错误严重程度
export enum ErrorSeverity {
  LOW = 'low',
  MEDIUM = 'medium',
  HIGH = 'high',
  CRITICAL = 'critical'
}

// 统一错误接口
export interface UnifiedError {
  code: string
  message: string
  type: ErrorType
  severity: ErrorSeverity
  timestamp: string
  requestId?: string
  details?: {
    field?: string
    message?: string
    code?: string
  }[] | string
  extraData?: Record<string, unknown>
}

// 错误处理选项
export interface ErrorHandlerOptions {
  showMessage?: boolean
  showNotification?: boolean
  logToConsole?: boolean
  customMessage?: string
  onError?: (error: UnifiedError) => void
}

// 错误码映射
const ERROR_CODE_MAP: Record<string, { type: ErrorType; severity: ErrorSeverity; defaultMessage: string }> = {
  // 通用错误 (1000-1999)
  'E1000': { type: ErrorType.INTERNAL_SERVER_ERROR, severity: ErrorSeverity.CRITICAL, defaultMessage: '服务器内部错误' },
  'E1001': { type: ErrorType.API_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '无效请求' },
  'E1002': { type: ErrorType.VALIDATION_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '数据验证失败' },
  'E1003': { type: ErrorType.AUTHENTICATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '未授权访问' },
  'E1004': { type: ErrorType.AUTHORIZATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '权限不足' },
  'E1005': { type: ErrorType.API_ERROR, severity: ErrorSeverity.LOW, defaultMessage: '资源未找到' },

  // 业务错误 (2000-2999)
  'E2000': { type: ErrorType.BUSINESS_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '业务逻辑错误' },
  'E2001': { type: ErrorType.API_ERROR, severity: ErrorSeverity.LOW, defaultMessage: '资源未找到' },
  'E2002': { type: ErrorType.BUSINESS_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '资源已存在' },
  'E2003': { type: ErrorType.BUSINESS_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '资源冲突' },
  'E2004': { type: ErrorType.AUTHORIZATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '操作不被允许' },

  // 数据验证错误 (3000-3999)
  'E3000': { type: ErrorType.VALIDATION_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '数据验证失败' },
  'E3001': { type: ErrorType.VALIDATION_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '必填字段缺失' },
  'E3002': { type: ErrorType.VALIDATION_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '字段格式无效' },
  'E3003': { type: ErrorType.VALIDATION_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '字段值超出范围' },

  // 权限和认证错误 (4000-4999)
  'E4000': { type: ErrorType.AUTHENTICATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '认证失败' },
  'E4001': { type: ErrorType.AUTHENTICATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '登录已过期' },
  'E4002': { type: ErrorType.AUTHENTICATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '令牌无效' },
  'E4003': { type: ErrorType.AUTHORIZATION_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '权限不足' },

  // 外部服务错误 (5000-5999)
  'E5000': { type: ErrorType.INTERNAL_SERVER_ERROR, severity: ErrorSeverity.HIGH, defaultMessage: '外部服务错误' },
  'E5001': { type: ErrorType.INTERNAL_SERVER_ERROR, severity: ErrorSeverity.CRITICAL, defaultMessage: '数据库错误' },
  'E5002': { type: ErrorType.INTERNAL_SERVER_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: '文件上传失败' },
  'E5004': { type: ErrorType.INTERNAL_SERVER_ERROR, severity: ErrorSeverity.MEDIUM, defaultMessage: 'PDF处理失败' }
}

/**
 * 统一错误处理器类
 */
export class UnifiedErrorHandler {
  private static instance: UnifiedErrorHandler
  private errorQueue: UnifiedError[] = []
  private maxQueueSize = 100

  private constructor() {
    // 设置全局错误监听
    this.setupGlobalErrorHandlers()
  }

  static getInstance(): UnifiedErrorHandler {
    if (!UnifiedErrorHandler.instance) {
      UnifiedErrorHandler.instance = new UnifiedErrorHandler()
    }
    return UnifiedErrorHandler.instance
  }

  /**
   * 处理错误
   */
  handleError(error: unknown, options: ErrorHandlerOptions = {}): UnifiedError {
    const unifiedError = this.parseError(error)

    // 添加到错误队列
    this.addToErrorQueue(unifiedError)

    // 日志记录
    if (options.logToConsole !== false) {
      this.logError(unifiedError)
    }

    // 显示错误消息
    if (options.showMessage !== false) {
      this.showErrorMessage(unifiedError, options)
    }

    // 调用自定义回调
    if (options.onError) {
      options.onError(unifiedError)
    }

    return unifiedError
  }

  /**
   * 解析各种类型的错误为统一错误格式
   */
  private parseError(error: unknown): UnifiedError {
    // Axios错误
    if (this.isAxiosError(error)) {
      return this.parseAxiosError(error)
    }

    // 统一错误格式
    if (this.isUnifiedError(error)) {
      return error
    }

    // 标准Error对象
    if (error instanceof Error) {
      return this.parseStandardError(error)
    }

    // 字符串错误
    if (typeof error === 'string') {
      return this.parseStringError(error)
    }

    // 未知错误
    return this.parseUnknownError(error)
  }

  /**
   * 解析Axios错误
   */
  private parseAxiosError(error: AxiosError): UnifiedError {
    const response = error.response
    const request = error.request

    // 网络错误
    if (!response && request) {
      return {
        code: 'NETWORK_ERROR',
        message: '网络连接失败，请检查网络设置',
        type: ErrorType.NETWORK_ERROR,
        severity: ErrorSeverity.MEDIUM,
        timestamp: new Date().toISOString(),
        details: error.message
      }
    }

    // 服务器响应错误
    if (response?.data) {
      const data = response.data as any

      // 后端统一错误格式
      if (data.error && data.error.code) {
        const errorInfo = ERROR_CODE_MAP[data.error.code] || {
          type: ErrorType.UNKNOWN_ERROR,
          severity: ErrorSeverity.MEDIUM,
          defaultMessage: '未知错误'
        }

        return {
          code: data.error.code,
          message: data.error.message || errorInfo.defaultMessage,
          type: errorInfo.type,
          severity: errorInfo.severity,
          timestamp: data.error.timestamp || new Date().toISOString(),
          requestId: data.request_id,
          details: data.error.details,
          extraData: {
            status: response.status,
            statusText: response.statusText,
            url: response.config.url
          }
        }
      }

      // 其他格式的错误响应
      return {
        code: `HTTP_${response.status}`,
        message: data.message || data.error?.message || `请求失败 (${response.status})`,
        type: response.status >= 500 ? ErrorType.INTERNAL_SERVER_ERROR : ErrorType.API_ERROR,
        severity: response.status >= 500 ? ErrorSeverity.HIGH : ErrorSeverity.MEDIUM,
        timestamp: new Date().toISOString(),
        extraData: {
          status: response.status,
          statusText: response.statusText,
          url: response.config.url,
          responseData: data
        }
      }
    }

    // 其他Axios错误
    return {
      code: 'AXIOS_ERROR',
      message: error.message || '请求失败',
      type: ErrorType.API_ERROR,
      severity: ErrorSeverity.MEDIUM,
      timestamp: new Date().toISOString(),
      details: error.message
    }
  }

  /**
   * 解析标准Error对象
   */
  private parseStandardError(error: Error): UnifiedError {
    return {
      code: 'STANDARD_ERROR',
      message: error.message,
      type: ErrorType.UNKNOWN_ERROR,
      severity: ErrorSeverity.MEDIUM,
      timestamp: new Date().toISOString(),
      details: error.stack
    }
  }

  /**
   * 解析字符串错误
   */
  private parseStringError(error: string): UnifiedError {
    return {
      code: 'STRING_ERROR',
      message: error,
      type: ErrorType.UNKNOWN_ERROR,
      severity: ErrorSeverity.MEDIUM,
      timestamp: new Date().toISOString()
    }
  }

  /**
   * 解析未知错误
   */
  private parseUnknownError(error: unknown): UnifiedError {
    return {
      code: 'UNKNOWN_ERROR',
      message: '未知错误',
      type: ErrorType.UNKNOWN_ERROR,
      severity: ErrorSeverity.MEDIUM,
      timestamp: new Date().toISOString(),
      details: typeof error === 'object' ? JSON.stringify(error) : String(error)
    }
  }

  /**
   * 显示错误消息
   */
  private showErrorMessage(error: UnifiedError, options: ErrorHandlerOptions): void {
    const displayMessage = options.customMessage || error.message

    // CRITICAL错误使用通知
    if (error.severity === ErrorSeverity.CRITICAL) {
      notification.error({
        message: '严重错误',
        description: displayMessage,
        duration: 0, // 不自动关闭
        key: error.code
      })
      return
    }

    // HIGH错误使用持久通知
    if (error.severity === ErrorSeverity.HIGH) {
      if (options.showNotification !== false) {
        notification.error({
          message: '错误',
          description: displayMessage,
          duration: 8,
          key: error.code
        })
      }
      return
    }

    // MEDIUM和LOW错误使用message
    if (options.showMessage !== false) {
      message.error(displayMessage)
    }
  }

  /**
   * 记录错误日志
   */
  private logError(error: UnifiedError): void {
    const logData = {
      ...error,
      userAgent: navigator.userAgent,
      url: window.location.href,
      timestamp: new Date().toISOString()
    }

    switch (error.severity) {
      case ErrorSeverity.CRITICAL:
        console.error('🚨 Critical Error:', logData)
        break
      case ErrorSeverity.HIGH:
        console.error('🔴 High Severity Error:', logData)
        break
      case ErrorSeverity.MEDIUM:
        console.warn('🟡 Medium Severity Error:', logData)
        break
      case ErrorSeverity.LOW:
        console.info('🟢 Low Severity Error:', logData)
        break
    }
  }

  /**
   * 添加到错误队列
   */
  private addToErrorQueue(error: UnifiedError): void {
    this.errorQueue.push(error)
    if (this.errorQueue.length > this.maxQueueSize) {
      this.errorQueue.shift()
    }
  }

  /**
   * 获取错误队列
   */
  getErrorQueue(): UnifiedError[] {
    return [...this.errorQueue]
  }

  /**
   * 清空错误队列
   */
  clearErrorQueue(): void {
    this.errorQueue = []
  }

  /**
   * 设置全局错误监听
   */
  private setupGlobalErrorHandlers(): void {
    // 全局未捕获的错误
    window.addEventListener('error', (event) => {
      this.handleError(event.error || new Error(event.message), {
        showMessage: true,
        logToConsole: true
      })
    })

    // 全局未捕获的Promise错误
    window.addEventListener('unhandledrejection', (event) => {
      this.handleError(event.reason, {
        showMessage: true,
        logToConsole: true
      })
    })
  }

  /**
   * 类型检查方法
   */
  private isAxiosError(error: unknown): error is AxiosError {
    return error !== null && typeof error === 'object' && 'isAxiosError' in error
  }

  private isUnifiedError(error: unknown): error is UnifiedError {
    return error !== null && typeof error === 'object' &&
           'code' in error && 'message' in error && 'type' in error && 'severity' in error
  }
}

// 创建全局实例
export const unifiedErrorHandler = UnifiedErrorHandler.getInstance()

// 便捷的错误处理函数
export const handleError = (error: unknown, options?: ErrorHandlerOptions) => {
  return unifiedErrorHandler.handleError(error, options)
}

// 特定类型的错误处理函数
export const handleNetworkError = (error: unknown, options?: ErrorHandlerOptions) => {
  return handleError(error, {
    customMessage: '网络连接失败，请检查网络设置后重试',
    ...options
  })
}

export const handleAuthError = (error: unknown, onRedirect?: () => void) => {
  const unifiedError = handleError(error, {
    customMessage: '登录已过期，请重新登录',
    showMessage: true
  })

  // 如果是认证错误，执行重定向
  if (unifiedError.type === ErrorType.AUTHENTICATION_ERROR && onRedirect) {
    setTimeout(onRedirect, 2000)
  }

  return unifiedError
}

export const handleValidationError = (error: unknown, options?: ErrorHandlerOptions) => {
  return handleError(error, {
    customMessage: '数据验证失败，请检查输入内容',
    ...options
  })
}

export const handleBusinessError = (error: unknown, options?: ErrorHandlerOptions) => {
  return handleError(error, options)
}

export default unifiedErrorHandler

File: backend/src/core/unified_error_handler.py
"""
统一错误处理系统
整合所有错误处理机制，提供一致的错误处理体验
"""

import logging
import traceback
from datetime import datetime
from enum import Enum
from typing import Any

from fastapi import HTTPException, Request, status
from fastapi.responses import JSONResponse
from pydantic import BaseModel


class ErrorCode(str, Enum):
    """统一错误码枚举"""

    # 通用错误 (1000-1999)
    INTERNAL_SERVER_ERROR = "E1000"
    INVALID_REQUEST = "E1001"
    VALIDATION_ERROR = "E1002"
    UNAUTHORIZED = "E1003"
    FORBIDDEN = "E1004"
    NOT_FOUND = "E1005"
    METHOD_NOT_ALLOWED = "E1006"
    TIMEOUT = "E1007"

    # 业务错误 (2000-2999)
    BUSINESS_ERROR = "E2000"
    RESOURCE_NOT_FOUND = "E2001"
    RESOURCE_ALREADY_EXISTS = "E2002"
    RESOURCE_CONFLICT = "E2003"
    OPERATION_NOT_ALLOWED = "E2004"
    INVALID_OPERATION = "E2005"

    # 数据验证错误 (3000-3999)
    DATA_VALIDATION_ERROR = "E3000"
    REQUIRED_FIELD_MISSING = "E3001"
    INVALID_FIELD_FORMAT = "E3002"
    FIELD_VALUE_OUT_OF_RANGE = "E3003"
    INVALID_DATE_FORMAT = "E3004"

    # 权限和认证错误 (4000-4999)
    AUTHENTICATION_FAILED = "E4000"
    TOKEN_EXPIRED = "E4001"
    TOKEN_INVALID = "E4002"
    INSUFFICIENT_PERMISSIONS = "E4003"
    ACCOUNT_LOCKED = "E4004"
    ACCOUNT_DISABLED = "E4005"

    # 外部服务错误 (5000-5999)
    EXTERNAL_SERVICE_ERROR = "E5000"
    DATABASE_ERROR = "E5001"
    FILE_UPLOAD_ERROR = "E5002"
    EMAIL_SERVICE_ERROR = "E5003"
    PDF_PROCESSING_ERROR = "E5004"


class ErrorSeverity(str, Enum):
    """错误严重程度"""

    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class ErrorDetail(BaseModel):
    """错误详情模型"""

    field: str | None = None
    message: str
    code: str | None = None


class UnifiedErrorResponse(BaseModel):
    """统一错误响应模型"""

    success: bool = False
    error: dict[str, Any]
    timestamp: datetime
    request_id: str | None = None


class UnifiedError(Exception):
    """统一错误异常类"""

    def __init__(
        self,
        message: str,
        code: ErrorCode = ErrorCode.BUSINESS_ERROR,
        status_code: int = status.HTTP_400_BAD_REQUEST,
        details: str | dict[str, Any] | list | None = None,
        severity: ErrorSeverity = ErrorSeverity.MEDIUM,
        extra_data: dict[str, Any] | None = None,
    ):
        self.message = message
        self.code = code
        self.status_code = status_code
        self.details = details
        self.severity = severity
        self.extra_data = extra_data or {}
        super().__init__(self.message)

    def to_dict(self) -> dict[str, Any]:
        """转换为字典格式"""
        error_data = {
            "code": self.code.value,
            "message": self.message,
            "severity": self.severity.value,
            "timestamp": datetime.utcnow().isoformat(),
            **self.extra_data,
        }

        if self.details:
            if isinstance(self.details, str):
                error_data["details"] = self.details
            elif isinstance(self.details, dict):
                error_data.update(self.details)
            elif isinstance(self.details, list):
                error_data["field_errors"] = self.details

        return error_data


class UnifiedErrorHandler:
    """统一错误处理器"""

    def __init__(self, logger: logging.Logger | None = None):
        self.logger = logger or logging.getLogger(__name__)
        self._setup_logger()

    def _setup_logger(self):
        """设置日志格式"""
        if not self.logger.handlers:
            handler = logging.StreamHandler()
            formatter = logging.Formatter(
                "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
            )
            handler.setFormatter(formatter)
            self.logger.addHandler(handler)
            self.logger.setLevel(logging.INFO)

    def handle_error(
        self, error: Exception, request: Request | None = None
    ) -> JSONResponse:
        """处理错误并返回统一响应"""

        request_id = getattr(request.state, "request_id", None) if request else None

        # 记录错误日志
        self._log_error(error, request)

        # 根据错误类型创建响应
        if isinstance(error, UnifiedError):
            error_data = error.to_dict()
            status_code = error.status_code
        elif isinstance(error, HTTPException):
            error_data = self._handle_http_exception(error)
            status_code = error.status_code
        elif isinstance(error, ValueError):
            error_data = self._handle_validation_error(error)
            status_code = status.HTTP_422_UNPROCESSABLE_ENTITY
        else:
            error_data = self._handle_unknown_error(error)
            status_code = status.HTTP_500_INTERNAL_SERVER_ERROR

        # 添加请求ID
        if request_id:
            error_data["request_id"] = request_id

        response = UnifiedErrorResponse(
            error=error_data, timestamp=datetime.utcnow(), request_id=request_id
        )

        return JSONResponse(
            status_code=status_code, content=response.model_dump(mode="json")
        )

    def _log_error(self, error: Exception, request: Request | None = None):
        """记录错误日志"""
        error_info = {
            "error_type": type(error).__name__,
            "error_message": str(error),
            "timestamp": datetime.utcnow().isoformat(),
        }

        if request:
            error_info.update(
                {
                    "method": request.method,
                    "url": str(request.url),
                    "client_ip": request.client.host if request.client else None,
                    "user_agent": request.headers.get("user-agent"),
                }
            )

        # 根据错误严重程度选择日志级别
        if isinstance(error, UnifiedError):
            if error.severity == ErrorSeverity.CRITICAL:
                self.logger.critical("Critical error occurred", extra=error_info)
            elif error.severity == ErrorSeverity.HIGH:
                self.logger.error("High severity error occurred", extra=error_info)
            elif error.severity == ErrorSeverity.MEDIUM:
                self.logger.warning("Medium severity error occurred", extra=error_info)
            else:
                self.logger.info("Low severity error occurred", extra=error_info)
        else:
            self.logger.error("Unexpected error occurred", extra=error_info)

        # 记录详细的堆栈跟踪
        if not isinstance(error, (UnifiedError, HTTPException)):
            self.logger.debug(f"Error traceback: {traceback.format_exc()}")

    def _handle_http_exception(self, error: HTTPException) -> dict[str, Any]:
        """处理HTTP异常"""
        return {
            "code": f"E{error.status_code}",
            "message": error.detail,
            "severity": ErrorSeverity.MEDIUM.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _handle_validation_error(self, error: ValueError) -> dict[str, Any]:
        """处理验证错误"""
        return {
            "code": ErrorCode.DATA_VALIDATION_ERROR.value,
            "message": "数据验证失败",
            "details": str(error),
            "severity": ErrorSeverity.MEDIUM.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _handle_unknown_error(self, error: Exception) -> dict[str, Any]:
        """处理未知错误"""
        return {
            "code": ErrorCode.INTERNAL_SERVER_ERROR.value,
            "message": "服务器内部错误",
            "details": str(error) if self._is_development() else None,
            "severity": ErrorSeverity.HIGH.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _is_development(self) -> bool:
        """检查是否为开发环境"""
        import os

        return os.getenv("ENVIRONMENT", "development").lower() == "development"


# 创建全局错误处理器实例
unified_error_handler = UnifiedErrorHandler()


# 便捷的错误创建函数
def create_business_error(
    message: str,
    code: ErrorCode = ErrorCode.BUSINESS_ERROR,
    details: Any | None = None,
) -> UnifiedError:
    """创建业务错误"""
    return UnifiedError(
        message=message,
        code=code,
        status_code=status.HTTP_400_BAD_REQUEST,
        details=details,
        severity=ErrorSeverity.MEDIUM,
    )


def create_not_found_error(
    message: str = "资源未找到", resource_type: str | None = None
) -> UnifiedError:
    """创建资源未找到错误"""
    extra_data = {}
    if resource_type:
        extra_data["resource_type"] = resource_type

    return UnifiedError(
        message=message,
        code=ErrorCode.RESOURCE_NOT_FOUND,
        status_code=status.HTTP_404_NOT_FOUND,
        severity=ErrorSeverity.LOW,
        extra_data=extra_data,
    )


def create_validation_error(
    message: str, field_errors: list | None = None
) -> UnifiedError:
    """创建验证错误"""
    return UnifiedError(
        message=message,
        code=ErrorCode.DATA_VALIDATION_ERROR,
        status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
        details=field_errors,
        severity=ErrorSeverity.MEDIUM,
    )


def create_authentication_error(message: str = "认证失败") -> UnifiedError:
    """创建认证错误"""
    return UnifiedError(
        message=message,
        code=ErrorCode.AUTHENTICATION_FAILED,
        status_code=status.HTTP_401_UNAUTHORIZED,
        severity=ErrorSeverity.HIGH,
    )


def create_authorization_error(message: str = "权限不足") -> UnifiedError:
    """创建授权错误"""
    return UnifiedError(
        message=message,
        code=ErrorCode.INSUFFICIENT_PERMISSIONS,
        status_code=status.HTTP_403_FORBIDDEN,
        severity=ErrorSeverity.HIGH,
    )


def create_internal_error(
    message: str = "服务器内部错误", original_error: Exception | None = None
) -> UnifiedError:
    """创建内部服务器错误"""
    extra_data = {}
    if original_error:
        extra_data["original_error"] = str(original_error)

    return UnifiedError(
        message=message,
        code=ErrorCode.INTERNAL_SERVER_ERROR,
        status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
        severity=ErrorSeverity.CRITICAL,
        extra_data=extra_data,
    )


class ErrorHandler:
    """统一错误处理器类"""

    def __init__(self, logger_name: str = "unified_error_handler"):
        self.logger = logging.getLogger(logger_name)
        self.error_handlers = {
            UnifiedError: self._handle_unified_error,
            HTTPException: self._handle_http_exception,
            ValueError: self._handle_validation_error,
            Exception: self._handle_unknown_error,
        }

    def handle_error(
        self, error: Exception, context: dict[str, Any] | None = None
    ) -> dict[str, Any]:
        """处理错误并返回标准化响应"""
        error_type = type(error)
        handler = self.error_handlers.get(error_type, self._handle_unknown_error)

        error_response = handler(error)

        if context:
            error_response["context"] = context

        # 记录错误
        self._log_error(error, error_response)

        return error_response

    def _handle_unified_error(self, error: UnifiedError) -> dict[str, Any]:
        """处理统一错误"""
        return {
            "code": error.code.value,
            "message": error.message,
            "details": error.details,
            "severity": error.severity.value,
            "timestamp": datetime.utcnow().isoformat(),
            "extra_data": error.extra_data,
        }

    def _handle_http_exception(self, error: HTTPException) -> dict[str, Any]:
        """处理HTTP异常"""
        return {
            "code": f"E{error.status_code}",
            "message": error.detail,
            "severity": ErrorSeverity.MEDIUM.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _handle_validation_error(self, error: ValueError) -> dict[str, Any]:
        """处理验证错误"""
        return {
            "code": ErrorCode.DATA_VALIDATION_ERROR.value,
            "message": "数据验证失败",
            "details": str(error),
            "severity": ErrorSeverity.MEDIUM.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _handle_unknown_error(self, error: Exception) -> dict[str, Any]:
        """处理未知错误"""
        return {
            "code": ErrorCode.INTERNAL_SERVER_ERROR.value,
            "message": "服务器内部错误",
            "details": str(error) if self._is_development() else None,
            "severity": ErrorSeverity.HIGH.value,
            "timestamp": datetime.utcnow().isoformat(),
        }

    def _log_error(self, error: Exception, error_response: dict[str, Any]):
        """记录错误日志"""
        error_info = {
            "error_type": type(error).__name__,
            "error_code": error_response["code"],
            "error_message": error_response["message"],
            "severity": error_response["severity"],
        }

        if isinstance(error, UnifiedError):
            if error.severity == ErrorSeverity.CRITICAL:
                self.logger.critical("Critical error occurred", extra=error_info)
            elif error.severity == ErrorSeverity.HIGH:
                self.logger.error("High severity error occurred", extra=error_info)
            elif error.severity == ErrorSeverity.MEDIUM:
                self.logger.warning("Medium severity error occurred", extra=error_info)
            else:
                self.logger.info("Low severity error occurred", extra=error_info)
        else:
            self.logger.error("Unexpected error occurred", extra=error_info)

    def _is_development(self) -> bool:
        """检查是否为开发环境"""
        import os

        return os.getenv("ENVIRONMENT", "development").lower() == "development"
